stl/inc/complex

```diff
@@ -697,14 +697,6 @@ public:
     constexpr complex(const _Fcomplex_value& _Right)
         : _Complex_base<float, _Fcomplex_value>(_Right._Val[_RE], _Right._Val[_IM]) {}
 
-    constexpr complex(const _Dcomplex_value& _Right)
-        : _Complex_base<float, _Fcomplex_value>(
-            static_cast<float>(_Right._Val[_RE]), static_cast<float>(_Right._Val[_IM])) {}
-
-    constexpr complex(const _Lcomplex_value& _Right)
-        : _Complex_base<float, _Fcomplex_value>(
-            static_cast<float>(_Right._Val[_RE]), static_cast<float>(_Right._Val[_IM])) {}

```

Comment:
Very nice! Thanks for finding this simple fix! 😸

This makes me wonder - can we remove `complex<float>`'s remaining constructor from `const _Fcomplex_value&`, and likewise for `complex<double>` and `complex<long double>`? It appears that they're never actually invoked. The `_Fcomplex_value` etc. types need to remain in the inheritance hierarchy to be used as storage (for bincompat) but if we can remove the constructors without triggering compiler errors, that would be a further simplification and compiler throughput improvement.

Reply:
Honestly, this fix is inspired by @MattStephanson comment on the issue page https://github.com/microsoft/STL/issues/1256#issuecomment-687684231